### ISSUE TITLE
gblinear for skewed classification problem

### ISSUE THREAD

**Issue by bingojojstu** [dmlc/xgboost#1705]
Hi All, 

I am a newbie in machine learning and I have read most of the questions here with no satisfying answers.

Hope this post won't offend anyone here.

Here it goes:

I have a highly unbalanced data set with 0.4% Positive Flags out of a total of ~50,000 data sample (that gives roughly 200 positive flag) with ~700 attributes.

I did Principal component analysis to reduce the co-variance and then dump the model into xgboost using gblinear for binary logistic regression. The detail param are {eval_metric:'auc',max_depth:6, alpha:0.0001,lambda:1,lambda_bias:1}

I did a 64:16:20 stratified split for Train, Validation and test set.

I used the train and validation set to train the model with num_rounds=500 and early stopping rounds=10.   After that, I predict on the test set.

The problem is the following: I get decent segregation power (auc is around 0.55 and kolmogorov smirnov test score is about 30%).

But when I compare the mean value of Y_predicted_probability to the true average, it is totally off (i.e. Y_predicted_probability.mean() = 0.55% instead of around 0.4%).

After that, I used feature selection from sklearn (f_classif) to reduce the number of attributes down to ~40 and followed by training the model. 

The prediction capability doesn't drop much and the mean value of predicted probability is also spot on.

Could any one point me to the right direction as of why this happens? 

I am really confused because my understanding about bias and variance is exactly the opposite.

Thanks in advance.

**Comment by pommedeterresautee** [comment]
There is no depth for a linear model!
You can reweight your positive observations when building your DMatrix or check the documentation for parameters to make the reweighting even easier.
